userguide/storage.rst

```diff
@@ -713,53 +713,64 @@ existing datasets to close them--the remaining
-turning this option on.
-
-If deduplication is changed to *On*, duplicate data blocks are removed
-synchronously. The result is that only unique data is stored and
-common components are shared among files. If deduplication is changed
-to *Verify*, ZFS will do a byte-to-byte comparison when two blocks
-have the same signature to make sure that the block contents are
-identical. Since hash collisions are extremely rare, *Verify* is
-usually not worth the performance hit.
-
-.. note:: After deduplication is enabled, the only way to disable it
-   is to use the :samp:`zfs set dedup=off {dataset_name}` command
-   from :ref:`Shell`. However, any data that has already been
-   deduplicated will not be un-deduplicated. Only newly stored data
-   after the property change will not be deduplicated. The only way to
-   remove existing deduplicated data is to copy all of the data off of
-   the dataset, set the property to off, then copy the data back in
-   again. Alternately, create a new dataset with
-   :guilabel:`ZFS Deduplication` left disabled, copy the data to the
-   new dataset, and destroy the original dataset.
+Deduplication (often just called "dedup") is the process of storing 
+just one copy of any data that is found to be the same in different 
+files. The aim is that only unique data is physically stored, rather
+than another identical copy being stored for every file which contains 
+that data. Deduplication potentially allows a large number of very 
+similar files to be held very efficiently within a much smaller 
+amount of storage capacity, if the data contains numerous identical 
+files or blocks of data.  However, deduplication also places a very 
+heavy burden on RAM and on the CPU, and in particular, deduplication 
+can significantly slow down data write performance and usually requires
+large (and at times huge) amounts of RAM. See 
+:ref:`Deduplication RAM requirements` for details.
+
+.. warning::
+
+   The more data written to a deduplicated dataset, the more RAM it 
+   requires. The DDTs (dedup tables) also compete with other data for RAM,
+   and can be pushed out of RAM in some cases. When the system starts 
+   storing DDTs on disk because they no longer fit into RAM or have been
+   pushed out of RAM, performance craters.
+   
+   Importing an unclean deduplicated pool can also require large amounts of RAM, 
+   so that the deduplicated data in the pool can be checked during import. 
+   If the system does not have the needed RAM, it will panic. The only 
+   solution then is to add more RAM or recreate the pool.
+
+   **There is also no quick way to undedup data 
+   within a dataset once deduplication is enabled**, as disabling 
+   deduplication has **NO EFFECT** on existing data. The simplest way to 
+   undedup existing data is to copy the deduped files to a new undeduped 
+   dataset or location in the pool or elsewhere, delete the original 
+   data, and (if necessary) copy back the undeduped copies of the data.  
```

Comment:
Lots of asides in this paragraph, which interrupt the flow of sentences and distract from the primary point.  Move asides to separate sentences.

Reply:
updated, please re-review